src/OrleansTestingHost/TestingSiloHost.cs

```diff
@@ -630,6 +694,11 @@ private static void StopOrleansSilo(SiloHandle instance, bool stopGracefully)
                 }
             }
 
+            // give silo a bit more time before unloading appdomain
+            // e.g. for completing async storage requests
+            if (stopGracefully)
```

Comment:
this does not sound like a good idea.
We can hook into Gracefull shutdown, wait until silo shuts down and do all "async storage requests" in the providers stop (providers stop is called as part of Gracefull shutdown and silo host can waat for silo to signal Gracefull shutdown was done).